src/core/libraries/np_trophy/np_trophy.cpp

```diff
@@ -249,19 +257,8 @@ int PS4_SYSV_ABI sceNpTrophyGetGameInfo(OrbisNpTrophyContext context, OrbisNpTro
 
     if (result) {
 
-        uint32_t numGroups = 0;
-        uint32_t numTrophies = 0;
-        uint32_t numTrophiesByRarity[5];
-        numTrophiesByRarity[1] = 0;
-        numTrophiesByRarity[2] = 0;
-        numTrophiesByRarity[3] = 0;
-        numTrophiesByRarity[4] = 0;
-        uint32_t unlockedTrophies = 0;
-        uint32_t unlockedTrophiesByRarity[5];
-        unlockedTrophiesByRarity[1] = 0;
-        unlockedTrophiesByRarity[2] = 0;
-        unlockedTrophiesByRarity[3] = 0;
-        unlockedTrophiesByRarity[4] = 0;
+        GameTrophyInfo gameInfo;
+        memset(&gameInfo, 0, sizeof(GameTrophyInfo));
 
         auto trophyconf = doc.child("trophyconf");
         for (pugi::xml_node_iterator it = trophyconf.children().begin();
```

Comment:
```suggestion
        for (const auto& node : trophyconf.children()) {
```
No need for iterators here. Additionally I see a lot of unnecessary std::string construction here which results in a lot of heap allocations. Consider using sting_view which has no additional cost.
```
const auto name = std::string_view{node.name()};